doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -1460,12 +1476,26 @@ inhabitant of the type) is interactively built using tactics. The interactive
 proof mode is described in Chapter :ref:`proofhandling` and the tactics in
 Chapter :ref:`Tactics`. The basic assertion command is:
 
-.. cmd:: Theorem @ident {? @binders } : @type
+.. cmd:: @thm_token @ident_decl {* @binder } : @term
+   :name: Theorem; Lemma; Fact; Remark; Corollary; Proposition
+
+   .. ; Property
+
+   .. insertprodn thm_token thm_token
+
+   .. prodn::
+      thm_token ::= Theorem
+      | Lemma
+      | Fact
+      | Remark
+      | Corollary
+      | Proposition
+      | Property
 
    After the statement is asserted, Coq needs a proof. Once a proof of
-   :token:`type` under the assumptions represented by :token:`binders` is given and
+   :n:`@type` under the assumptions represented by :n:`@binders` is given and
    validated, the proof is generalized into a proof of :n:`forall @binders, @type` and
-   the theorem is bound to the name :token:`ident` in the environment.
+   the theorem is bound to the name :n:`@ident` in the environment.
 
    .. exn:: The term @term has type @type which should be Set, Prop or Type.
```

Comment:
OTOH this error message is consistent with the current state of the assertion syntax.